L1.py

```diff
@@ -1,225 +1,279 @@
-        'X-CSRFToken' : self.csrftoken,
-        'Authorization' : 'Token bd9ecc45b397f0e22bd0a8bd3304010ab4b586d8',
-        'Cookie':f'csrftoken={self.csrftoken}; sessionid={self.sessionId}'
-    }
-        return self.post_nutritionplan_headers
-    
-    def func_post_meal_headers(self):
-        self.post_meal_headers = {
-        'Content-Type': 'application/json',
-        'Referer': self.post_mealReferer,
-        'X-CSRFToken' : self.csrftoken,
-        'Authorization' : 'Token bd9ecc45b397f0e22bd0a8bd3304010ab4b586d8',
-        'Cookie':f'csrftoken={self.csrftoken}; sessionid={self.sessionId}'
-    }
-        return self.post_meal_headers
-    
-    def func_post_mealitem_headers(self):
-        self.post_mealitem_headers = {
+
+    #baseReferers
+
+    base_workoutReferer = "https://wger.de/en/workout/"
+    base_nutritionReferer = "https://wger.de/en/nutrition/"
+
+
+    def __init__(self):
+        cookies = self.login_and_cookies()
+        self.csrftoken = cookies[0]
+        self.sessionId = cookies[1]
+
+        self.headers = {
         'Content-Type': 'application/json',
-        'Referer': self.post_mealitemReferer,
+        'Referer': '',
         'X-CSRFToken' : self.csrftoken,
         'Authorization' : 'Token bd9ecc45b397f0e22bd0a8bd3304010ab4b586d8',
         'Cookie':f'csrftoken={self.csrftoken}; sessionid={self.sessionId}'
     }
-        return self.post_mealitem_headers
-
-    #payloads
-
-    payload_training = {
-    'training':280895,
-    'description':'ssssssssssss',
-    'day':[6,7]
-}
-    payload_exercise = {
-    "exerciseday":142728,
-    "exercises":279
-}
-
-    payload_meal = {
-        "plan":75441
-}
-
-    payload_mealitem = {
-    "meal":188373,
-    "ingredient":9842,
-    "amount":200
-}
-
-    def __init__(self):
-        s.get(self.loginURL)
-        self.csrftoken = s.cookies['csrftoken']
-        self.sessionId = s.cookies['sessionid']
 
-        self.payload = {
+    @classmethod
+    def show_request_details(cls, response_list):
+        print(response_list[1])
+        print(f"status code: {response_list[0].status_code}")
+        print(f"response : {response_list[0].content}")
+        print("\n")
+    
+    def login_and_cookies(self):
+         self.s.get(self.loginURL)
+         csrftoken = self.s.cookies['csrftoken']
+         sessionId = self.s.cookies['sessionid']
+         self.payload = {
             'username': 'tiberiu1213',
             'password' : 'THeoTHEo2013&',
             'submit' : 'Login',
-            'csrfmiddlewaretoken':self.csrftoken
+            'csrfmiddlewaretoken':csrftoken
             }
+         self.loginResponse = self.s.post(self.loginURL,data = self.payload, headers=dict(Referer=self.loginURL))
+         return [csrftoken,sessionId]
+
+    #requests
 
-        self.loginResponse = s.post(self.loginURL,data = self.payload, headers=dict(Referer=self.loginURL))
+    def get(self, type, id = None):
+        if type == "workout":
```

Comment:
There's a VERY close relationship between the type and the suffix you add to the base_URL. Using this relationship you can get rid of the whole if/elif block.